tests/socketcall_32bit/test

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/perl
+
+use strict;
+
+use Test;
+BEGIN { plan tests => 3 }
+
+use File::Temp qw/ tempfile /;
+
+my $basedir = $0;
+$basedir =~ s|(.*)/[^/]*|$1|;
+
+###
+# functions
+
+sub key_gen {
+	my @chars = ("A".."Z", "a".."z");
+	my $key = "testsuite-" . time . "-";
+	$key .= $chars[rand @chars] for 1..8;
+	return $key;
+}
+
+###
+# setup
+
+# reset audit
+system("auditctl -D >& /dev/null");
+
+# create stdout/stderr sinks
+(my $fh_out, my $stdout) = tempfile(TEMPLATE => '/tmp/audit-testsuite-out-XXXX',
+				    UNLINK => 1);
+(my $fh_err, my $stderr) = tempfile(TEMPLATE => '/tmp/audit-testsuite-err-XXXX',
+				    UNLINK => 1);
+(my $fh_out2, my $stdout2) = tempfile(
+				TEMPLATE => '/tmp/audit-testsuite-out-XXXX',
+				UNLINK => 1);
+(my $fh_err2, my $stderr2) = tempfile(
+				TEMPLATE => '/tmp/audit-testsuite-err-XXXX',
+				UNLINK => 1);
+
+###
+# tests
+
+# set the socketcall filter
+my $key = key_gen();
+system("auditctl -a always,exit -F arch=b64 -S clone -S close -S creat -S dup -S dup2 -S dup3 -S execve -S exit -S exit_group -S fork -S open -S openat -S rename -S renameat -S unlink -S unlinkat -S vfork -S 288 -S accept -S connect -S listen -S socket -S socketpair -k $key");
+system("auditctl -a always,exit -F arch=b32 -S clone -S close -S creat -S dup -S dup2 -S dup3 -S execve -S exit -S exit_group -S fork -S open -S openat -S rename -S renameat -S unlink -S unlinkat -S vfork -k $key");
+# connect
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=3 -k $key");
+# listen
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=4 -k $key");
+# accept
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=5 -k $key");
+# socketpair
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=8 -k $key");
+# accept4
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=18 -k $key");
+
+# run the test
+system("echo Hello World. | $basedir/conn localhost 22");
+
+system("sleep 1;sync");
+
+# test if we generate any audit records from the filter rule
+my $result = system("ausearch -i -k $key > $stdout 2> $stderr");
+ok($result, 0);
+
+# test if we generate the SOCKETCALL record
+my $line;
+my $line2;
+my $type;
+my $id = "";
+my $found_socketcall = 0;
+my $found_sockaddr = 0;
+my $found_other = 0;
+while ($line = <$fh_out>) {
+	if ($line =~ /^type=SOCKETCALL /) {
+		$found_socketcall = 1;
+		#print $line;
+	} elsif ($line =~ /^type=SOCKADDR /) {
+		$found_sockaddr = 1;
+		if ($line =~ / : saddr=\{ fam=inet laddr=127.0.0.1 lport=22 \}/) {
+			($id) = ($line =~ / msg=audit\(.*:([0-9]*)\).* /);
+		}
+	} else {
+		$found_other += 1;
+	}
+}
+ok($found_socketcall);
+if($found_sockaddr && !$found_socketcall) {
+	seek($fh_out2, 0, 0);
+	system("ausearch -i -k $key -a $id > $stdout2 2> $stderr2");
+	while ($line2 = <$fh_out2>) {
+		print $line2;
+	}
+	print "Found other: $found_other\n";
```

Comment:
We've talked about similar things in the past when it comes to the audit-testsuite; I'd prefer to keep things minimal/clean in the tests.  Please remove stuff like this before submitting PRs/patches.